nemo_text_processing/g2p/modules.py

```diff
@@ -286,32 +293,40 @@ def __init__(
         """
         self.use_stresses = use_stresses
         self.set_graphemes_upper = set_graphemes_upper
+        self.phoneme_probability = phoneme_probability
+        self._rng = random.Random()
+
+        if not use_chars and self.phoneme_probability is not None:
+            self.use_chars = True
+            logging.warning(
+                "phoneme_probability was not None, characters will be enabled even though "
+                "use_chars was set to False."
+            )
+        else:
+            self.use_chars = use_chars
 
         if isinstance(phoneme_dict, str) or isinstance(phoneme_dict, pathlib.Path):
             # Load phoneme_dict from file path
-            self.phoneme_dict, self.symbols = self._parse_as_cmu_dict(
-                phoneme_dict,
-                use_stresses,
-                self.STRESS_SYMBOLS,
-                upper=set_graphemes_upper,
-            )
+            self.phoneme_dict, self.symbols = self._parse_as_cmu_dict(phoneme_dict)
         else:
             # Load phoneme_dict as dictionary object
             logging.info("Loading phoneme_dict as a Dict object. Extracting valid symbols from values.")
             self.phoneme_dict = phoneme_dict
 
             self.symbols = set()
-            for entries in phoneme_dict.values():
-                for phonemes in entries:
-                    self.symbols.update(phonemes)
+            for word, phonemes in phoneme_dict.items():
+                self.symbols.update(phonemes)
+                if self.use_chars:
+                    word_no_punct = self.PUNCT_REGEX.sub('', word)
+                    self.symbols.update(word_no_punct)
```

Comment:
This doesn't seem ideal to me. Looking through our current ipa dictionary we have entries like: `G6` and `DÉJÀ`. This results in errors with backwards-compatibility since we previously assumed that numbers [0-9] and loan characters (from French) are not consider English graphemes. We should either re-evaluate this logic or not process these entries when looping through the g2p dict.